src/OpenTelemetry/Trace/TracerProviderBuilderSdk.cs

```diff
@@ -24,7 +24,7 @@ namespace OpenTelemetry.Trace
     /// <summary>
     /// Build TracerProvider with Resource, Sampler, Processors and Instrumentation.
     /// </summary>
-    internal class TracerProviderBuilderSdk : TracerProviderBuilder
+    public class TracerProviderBuilderSdk : TracerProviderBuilder
```

Comment:
This is good and something that I think opens a lot of possibilities. I think what would be even better is to change up `TracerProviderSdk` to take factories for the processors just like it already does for the instrumentations. Should probably use use a factory for the sampler. If we made those 2 changes then we could actually split this all up into the config and the build phase and we could use the factory methods in the hosting version to call into the IServiceProvider to get those instances.

Reply:
Also, in general, there are too many parameters going into `TracerProviderSdk`. Would be better to create an options object that has all this stuff so that new things can be added without adding more and more parameters to the `TracerProviderSdk` ctor.

Reply:
Let me take a look and see what I can cook up. I think most of the stuff is `internal` in there so we should be able to change without breaking things.

Reply:
OK a new version has been pushed. I'm going to add some examples to the description.